src/Transformer/YamlTransformer.php

```diff
@@ -393,14 +396,25 @@ protected function getMatchingDocuments($pattern, $documents)
             if (isset($documents[$name])) {
                 return [$documents[$name]];
             }
-
             return [];
         }
 
-        // If we only have an astericks, we'll add all unnamed docs. By excluding named docs
-        // we don't run into a circular depndency issue.
-        if($pattern === '*') {
-            $pattern = 'anonymous-*';
+        // After="*" docs are after all documents except OTHER After="*",
+        // and likewise for Before="*"
+        if ($pattern === '*') {
+            return array_filter(
+                $documents,
+                function ($document) use ($flag) {
+                    if (empty($document['header'][$flag])) {
+                        return true;
+                    }
+                    $otherPatterns = $document['header'][$flag];
+                    if (is_array($otherPatterns)) {
+                        return !in_array('*', $otherPatterns);
+                    }
+                    return $otherPatterns !== '*';
+                }
+            );
```

Comment:
I can't comment on the actual line, but this is for line 431 (the todo). There's no sanitsation for the matching so any document name with strange chars will be passed into preg_replace and may break. Sorry, I never got around to fixing it.